### ISSUE TITLE
Toolbox item with no icon crashes IDE

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#3590]
```
FATAL ERROR [2018-01-08 18:27:14Z]: An unhandled exception has occured. Terminating Visual Studio? True
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget+<>c__DisplayClass46_0.<OnExposeEvent>b__1 (MonoDevelop.DesignerSupport.Toolbox.Category curCategory, MonoDevelop.DesignerSupport.Toolbox.Item item, Gdk.Size itemDimension) [0x000bd] in /Users/builder/data/lanes/5402/7d0c9d7a/source/monodevelop/main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/ToolboxWidget.cs:292 
  at MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget.IterateItems (MonoDevelop.DesignerSupport.Toolbox.Category category, System.Int32& xpos, System.Int32& ypos, MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget+ItemAction action) [0x00081] in /Users/builder/data/lanes/5402/7d0c9d7a/source/monodevelop/main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/ToolboxWidget.cs:849 
  at MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget.Iterate (System.Int32& xpos, System.Int32& ypos, MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget+CategoryAction catAction, MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget+ItemAction action) [0x000a2] in /Users/builder/data/lanes/5402/7d0c9d7a/source/monodevelop/main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/ToolboxWidget.cs:892 
  at MonoDevelop.DesignerSupport.Toolbox.ToolboxWidget.OnExposeEvent (Gdk.EventExpose e) [0x001cc] in /Users/builder/data/lanes/5402/7d0c9d7a/source/monodevelop/main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/ToolboxWidget.cs:246 
  at Gtk.Widget.exposeevent_cb (System.IntPtr widget, System.IntPtr evnt) [0x00016] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/gtk-sharp-None/gtk/generated/Widget.cs:1349 
2018-01-08 18:27:14.340 mono64[51735:8452508] *** Terminating app due to uncaught exception 'NSGenericException', reason: '-[NSAlert runModal] may not be invoked inside of transaction begin/commit pair, or inside of transaction commit (usually this means it was invoked inside of a view's -drawRect: method.)'
```

> VS bug #568060

**Comment by mhutch** [comment]
It also crashes if the image is set but doesn't have an explicit size.

**Comment by slluis** [comment]
Fixed in version 7.5.0.410 (master)

Pull Request #3864 merged by: Lluis Sanchez
Author: mono
Commit: 021536537d99aa4dee24e029e1029699691564cd (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #3864 merged by: Lluis Sanchez
Author: mono
Commit: 021536537d99aa4dee24e029e1029699691564cd (xamarin/monodevelop)
